src/test/java/seedu/clinic/logic/parser/AddCommandParserTest.java

```diff
@@ -64,37 +68,53 @@ public void parse_allFieldsPresent_success() {
-                + REMARK_DESC_BOB, new AddCommand(expectedSupplier));
-        assertParseSuccess(parser, NAME_DESC_WAREHOUSE_B2 + PHONE_DESC_WAREHOUSE_A + PHONE_DESC_WAREHOUSE_B
-                + ADDRESS_DESC_WAREHOUSE_B + REMARK_DESC_WAREHOUSE_B, new AddCommand(expectedWarehouse));
+        assertParseSuccess(parser, VALID_TYPE_SUPPLIER + NAME_DESC_BOB2 + PHONE_DESC_AMY + PHONE_DESC_BOB
+                + EMAIL_DESC_BOB + REMARK_DESC_BOB, new AddCommand(expectedSupplier));
+        assertParseSuccess(parser, VALID_TYPE_WAREHOUSE + NAME_DESC_WAREHOUSE_B2 + PHONE_DESC_WAREHOUSE_A
+                + PHONE_DESC_WAREHOUSE_B + ADDRESS_DESC_WAREHOUSE_B + REMARK_DESC_WAREHOUSE_B,
+                new AddCommand(expectedWarehouse));
 
         // multiple emails - last email accepted
-        assertParseSuccess(parser, NAME_DESC_BOB2 + PHONE_DESC_BOB + EMAIL_DESC_AMY + EMAIL_DESC_BOB
-                + REMARK_DESC_BOB, new AddCommand(expectedSupplier));
+        assertParseSuccess(parser, VALID_TYPE_SUPPLIER + NAME_DESC_BOB2 + PHONE_DESC_BOB + EMAIL_DESC_AMY
+                + EMAIL_DESC_BOB + REMARK_DESC_BOB, new AddCommand(expectedSupplier));
 
         // multiple addresses - last email accepted
-        assertParseSuccess(parser, NAME_DESC_WAREHOUSE_B2 + PHONE_DESC_WAREHOUSE_B + ADDRESS_DESC_WAREHOUSE_A
-                + ADDRESS_DESC_WAREHOUSE_B + REMARK_DESC_WAREHOUSE_B, new AddCommand(expectedWarehouse));
+        assertParseSuccess(parser, VALID_TYPE_WAREHOUSE + NAME_DESC_WAREHOUSE_B2 + PHONE_DESC_WAREHOUSE_B
+                + ADDRESS_DESC_WAREHOUSE_A + ADDRESS_DESC_WAREHOUSE_B + REMARK_DESC_WAREHOUSE_B,
+                new AddCommand(expectedWarehouse));
 
         // multiple remarks - last remark accepted
-        assertParseSuccess(parser, NAME_DESC_BOB2 + PHONE_DESC_BOB + EMAIL_DESC_BOB + REMARK_DESC_AMY
-                + REMARK_DESC_BOB, new AddCommand(expectedSupplier));
-        assertParseSuccess(parser, NAME_DESC_WAREHOUSE_B + PHONE_DESC_WAREHOUSE_B
+        assertParseSuccess(parser, VALID_TYPE_SUPPLIER + NAME_DESC_BOB2 + PHONE_DESC_BOB + EMAIL_DESC_BOB
+                + REMARK_DESC_AMY + REMARK_DESC_BOB, new AddCommand(expectedSupplier));
+        assertParseSuccess(parser, VALID_TYPE_WAREHOUSE + NAME_DESC_WAREHOUSE_B + PHONE_DESC_WAREHOUSE_B
                 + ADDRESS_DESC_WAREHOUSE_B + REMARK_DESC_WAREHOUSE_A + REMARK_DESC_WAREHOUSE_B,
                 new AddCommand(expectedWarehouse));
+
+        //all fields for suppliers
+        String userInputSupplier = " " + PREFIX_TYPE + "s " + NAME_DESC_BOB2 + PHONE_DESC_BOB
+                + EMAIL_DESC_BOB + REMARK_DESC_BOB;
+        AddCommand expectedCommandSupplier = new AddCommand(expectedSupplier);
+        assertParseSuccess(parser, userInputSupplier, expectedCommandSupplier);
+
+        //all fields for warehouses
+        String userInputWarehouse = " " + PREFIX_TYPE + "w " + NAME_DESC_WAREHOUSE_B2 + PHONE_DESC_WAREHOUSE_B
```

Comment:
I think theres  a type descriptor in the util also, so perhaps thats better to using the strings "w " "s "

Reply:
I've changed it to use one of the pre-created descriptor!